diagnostics/tropical_pacific_sea_level/tropical_pacific_sea_level.py

```diff
@@ -0,0 +1,671 @@
+modelfile = [[os.getenv('tauuo_file')],
+             [os.getenv('tauvo_file')],
+             [os.getenv('zos_file')]]
+areafile = os.getenv('areacello_file')
+path[Model_name[0]]=[modeldir,modelfile]
+
+Model_varname = [os.getenv('tauuo_var'),os.getenv('tauvo_var'),os.getenv('zos_var')]
+Model_dimname = [os.getenv('Model_dim0'),os.getenv('Model_dim1'),os.getenv('Model_dim2')]
+Model_coordname = [os.getenv('Model_coord0'),os.getenv('Model_coord1')]
+
+xname = Model_dimname[2]
+yname = Model_dimname[1]
+
+
+
+
+for nmodel,model in enumerate(Model_name):
+    modeldir = path[model][0]
+    modelfile = path[model][1]
+    multivar = []
+    for file in modelfile :
+        if len(file) == 1 :
+            multivar.append([os.path.join(modeldir,file[0])])
+        elif len(file) > 1 :
+            multifile = []
+            for ff in file :
+                multifile.append(os.path.join(modeldir,ff))
+            multivar.append(multifile)
+    modelin[model] = multivar
+
+#### create time axis (datatime.datetime)
+timeax = xr.cftime_range(start=cftime.datetime(ori_syear,1,1),end=cftime.datetime(ori_fyear,12,1),freq='MS')
+timeax = timeax.to_datetimeindex()    # cftime => datetime64
+
+print('--------------------------')
+print('Start processing model outputs')
+print('--------------------------')
+# initialization of dict and list  (!!!!!!!! remove all previous read model info if exec !!!!!!!!!!)
+nmodel = len(Model_name)
+nvar = len(Model_varname)
+
+ds_model_mlist = {}
+mean_mlist = {}
+season_mlist = {}
+linear_mlist = {}
+#### models
+for nmodel,model in enumerate(Model_name):
+    ds_model_list = {}
+    mean_list = {}
+    season_list = {}
+    linear_list = {}
+    for nvar,var in enumerate(Model_varname):
+        print('read %s %s'%(model,var))
+
+        print(modelin[model][nvar][0])
+        # read input data
+        ds_model = xr.open_dataset(modelin[model][nvar][0],use_cftime=True)
+
+
+        # crop data (time)
+        ds_model['time'] = timeax
+        da_model = ds_model[var].where((ds_model['time.year'] >= syear)&
+                                       (ds_model['time.year'] <= fyear)
+                                       ,drop=True)
+        # crop data (space)
+        da_model = da_model.where((ds_model.lat >= np.min(np.array(tp_lat_region)))&
+                                  (ds_model.lat <= np.max(np.array(tp_lat_region)))
+                                  ,drop=True)
+
+        # remove land value
+        da_model['lon'] = da_model.lon.where(da_model.lon<1000.,other=np.nan)
+        da_model['lat'] = da_model.lat.where(da_model.lat<1000.,other=np.nan)
+
+        # store all model data
+        ds_model_list[var] = da_model
+
+        # calculate mean
+        mean_list[var] = ds_model_list[var].mean(dim='time').compute()
+        ds_model_list[var] = ds_model_list[var]-mean_list[var]
+
+        # calculate seasonality
+        season_list[var] = ds_model_list[var].groupby('time.month').mean(dim='time').compute()
```

Comment:
Can you verify that this is doing proper weighting by the number of days in a month?  I recall some issue in the past where xarray treated all months as equal length.

Reply:
Right I think for groupby function in xarray the number of days in a month need to be weighted according to each month. However, the code is doing a month group meaning it is calculating the mean of all Jan, Feb, March....etc during the period of analysis. In addition, the variable here is monthly frequency output so do not have this problem.